### ISSUE TITLE
Device Error while training custom dataset

### ISSUE THREAD

**Issue by Caliber2** [fizyr/keras-maskrcnn#85]
InvalidArgumentError (see above for traceback): Cannot assign a device for operation 'training/Adam/gradients/filtered_detections/map/while/embedding_lookup_3_grad/Reshape_1/f_acc': Could not satisfy explicit device specification '' because the node was colocated with a group of nodes that required incompatible device '/job:localhost/replica:0/task:0/device:GPU:0'

**Comment by hgaiser** [comment]
I think I've seen this error before, I seem to remember updating tensorflow fixed this issue. Could you give that a shot? What version are you using at the moment?

**Comment by Caliber2** [comment]
@hgaiser Thanks for the reply. Anyways, just after posting the issue, I realized the combination CUDA 9, CUDNN 7, tensorflow 1.4 and tensorflow-gpu 1.8 solves it.
